lambda/healthCheckForCoAP.ts

```diff
@@ -0,0 +1,207 @@
+import { MetricUnits, logMetrics } from '@aws-lambda-powertools/metrics'
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb'
+import { SSMClient } from '@aws-sdk/client-ssm'
+import { InvokeCommand, LambdaClient } from '@aws-sdk/client-lambda'
+import middy from '@middy/core'
+import { fromEnv } from '@nordicsemiconductor/from-env'
+import assert from 'node:assert/strict'
+import { registerDevice } from '../devices/registerDevice.js'
+import { metricsForComponent } from './metrics/metrics.js'
+import { logger } from './util/logger.js'
+import { getAllAccountsSettings } from '../nrfcloud/allAccounts.js'
+import type { Settings } from '../nrfcloud/settings.js'
+import {
+	ValidateResponse,
+	checkMessageFromWebsocket,
+} from '../util/checkMessageFromWebsocket.js'
+import { createPublicKey, createPrivateKey } from 'node:crypto'
+
+const { DevicesTableName, stackName, websocketUrl, coapLambda } = fromEnv({
+	DevicesTableName: 'DEVICES_TABLE_NAME',
+	stackName: 'STACK_NAME',
+	websocketUrl: 'WEBSOCKET_URL',
+	coapLambda: 'COAP_LAMBDA',
+})(process.env)
+
+const log = logger('healthCheckForCoAP')
+const db = new DynamoDBClient({})
+const ssm = new SSMClient({})
+const lambda = new LambdaClient({})
+
+const { track, metrics } = metricsForComponent('healthCheckForCoAP')
+
+const allAccountsSettings = await getAllAccountsSettings({
+	ssm,
+	stackName,
+})()
+
+await Promise.all(
+	Object.entries(allAccountsSettings).map(async ([account, settings]) => {
+		if ('healthCheckSettings' in settings) {
+			await registerDevice({
+				db,
+				devicesTableName: DevicesTableName,
+			})({
+				id: settings.healthCheckSettings.healthCheckClientId,
+				model: settings.healthCheckSettings.healthCheckModel,
+				fingerprint: settings.healthCheckSettings.healthCheckFingerPrint,
+				account,
+			})
+		} else {
+			log.warn(`${account} does not have health check settings`)
```

Comment:
This means the system is broken and should not happen. Must be a critical error.

Reply:
I will log critical error here but not throw the error because it will be captured in the main function, then the metric will record fail.